### PR TITLE
Signing image builds

### PR DESCRIPTION
**Pull request by zvonkok**
- Enhanced the e2e test
- Added Connaisseur for e2e test
- New Changes: Simple workflow to sign images and push to a registry, 
- Overriding states via ConfigMap
- Added support for  helm pre-install and  helm post-install hooks
- Templates in the CR

### CODE REVIEW

File: test/undeploy/undeploy.go [hunk 1]
```diff
@@ -36,6 +37,10 @@ func main() {
 
 	cl := framework.NewControllerRuntimeClient()
 
+	framework.DeleteAllSpecialResources(cl)
```
Comment (zvonkok) [inline test/undeploy/undeploy.go:40]: @dagrayvid

File: test/framework/create_from_yaml.go [hunk 2]
```diff
@@ -41,7 +42,11 @@ func NewControllerRuntimeClient() client.Client {
 	})
```
Comment (dagrayvid) [inline test/framework/create_from_yaml.go:43]:
```suggestion
	       MetricsBindAddress: "0",
	})
```

I think that this change will disable metrics for this controller, which will help us avoid the test flake of
```2021-06-29T13:33:29.011-0400	INFO	exit  	OnError: framework.NewControllerRuntimeClient[create_from_yaml.go:42] unable to start manager: error listening on :8080: listen tcp :8080: bind: address already in use```

File: docs/debug.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
+# How to debug recipes
+
+The Special Resource Operatore (SRO) can read the states either from the local,
+http or a ConfigMap resource.
+
+To create the states in the template directory as a CM we can run the
+following:
+
+```bash
+VERSION=0.0.1 REPO=exammple SPECIALRESOURCE=multi-build make assets
```
Comment (dagrayvid) [inline docs/debug.md]:
```suggestion
VERSION=0.0.1 REPO=example SPECIALRESOURCE=multi-build make assets
```

File: docs/debug.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+```
+
+This command will create a CM with all the states and SRO will use them in the
+next reconcilation loop. This makes it easy to override manifests during
+development without rebuilding SRO.
+
+To update the CR one can run:
+
+```bash
+VERSION=0.0.1 REPO=exammple SPECIALRESOURCE=multi-build make
```
Comment (dagrayvid) [inline docs/debug.md]:
```suggestion
VERSION=0.0.1 REPO=example SPECIALRESOURCE=multi-build make
```

File: docs/debug.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+```
+
+For this to work one has to create a `kustomization.yaml` that will be processed
+during a make run. One cannot just override one state all states are either
+read from one resource. Here is an `kustomization.yaml` as an example from the
```
Comment (dagrayvid) [inline docs/debug.md]:
```suggestion
during a make run. One cannot just override one state, all states are
read from one source (http, local, or ConfigMap). Here is an `kustomization.yaml` as an example from the
```
Found this sentence confusing at first.

File: test/framework/create_from_yaml.go (hunk 2 above)
Comment (zvonkok) [reply test/framework/create_from_yaml.go:43]: This change is already in master. Resolving.

File: docs/debug.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+all the manifests, hooks and values printed on the console. Can be valuable for
+verifying if all `Values` are set and correclty interpreted.
+
+```yaml
+apiVersion: sro.openshift.io/v1beta1
+kind: SpecialResource
+metadata:
+  name: multi-build
+spec:
+  debug: false
```
Comment (SchSeba) [inline docs/debug.md:54]: I think this needs to be true in this example

File: docs/debug.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+http or a ConfigMap resource.
+
+To create the states in the template directory as a CM we can run the
+following:
+
+```bash
+VERSION=0.0.1 REPO=example SPECIALRESOURCE=multi-build make assets
+```
+
+This command will create a CM with all the states and SRO will use them in the
```
Comment (SchSeba) [inline docs/debug.md:13]: is this only for development or can be used in production also?
as the partner may not want to push the is assets into the SRO repo

File: manifests-k8s/0000_namespace_openshift-special-resource-operator.yaml [hunk 8]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  labels:
+    control-plane: controller-manager
+    openshift.io/cluster-monitoring: "true"
+  name: openshift-special-resource-operator
```
Comment (dagrayvid) [inline manifests-k8s/0000_namespace_openshift-special-resource-operator.yaml:7]: Do you want to add manifests-k8s to the PR? If yes, please rerun `PLATFORM=k8s make manifests` as they are not fully updated.

**Pull request merged**